sdk/core/azure-core-tracing-opentelemetry/README.md

```diff
@@ -27,58 +56,68 @@ documentation][OpenTelemetry] | [Samples][samples]
+Synchronously create a secret using [azure-security-keyvault-secrets][azure-security-keyvault-secrets] with tracing enabled.
 
-    The plugin package creates a logical span representing public API call to encapsulate all the underlying HTTP calls. By default OpenTelemetry 
-    `Context.current()` will be used as a parent context - check out [OpenTelemetry documentation](https://opentelemetry.io/docs/java/manual_instrumentation/#tracing) for more info.
-    Users can *optionally* pass the instance of `io.opentelemetry.context.Context` to the SDKs using key **PARENT_TRACE_CONTEXT_KEY** on the [Context][context] parameter of the calling method
-    to provide explicit parent context.
-    This [sample][sample_key_vault] provides an example when no user parent span is passed.
+The plugin package creates a logical span representing public API call to encapsulate all the underlying HTTP calls. By default OpenTelemetry
+`Context.current()` will be used as a parent context - check out [OpenTelemetry documentation](https://opentelemetry.io/docs/java/manual_instrumentation/#tracing) for more info.
+Users can *optionally* pass the instance of `io.opentelemetry.context.Context` to the SDKs using key **PARENT_TRACE_CONTEXT_KEY** on the [Context][context] parameter of the calling method
+to provide explicit parent context.
+This [sample][sample_key_vault] provides an example when parent span is picked up automatically.
 
 ```java
-// Get the Tracer Provider
-static TracerSdkProvider tracerProvider = OpenTelemetrySdk.getTracerProvider();
-private static final Tracer TRACER = configureOpenTelemetryAndLoggingExporter();
-
 public static void main(String[] args) {
-   doClientWork();
-}
-
-public static void doClientWork() {
+    configureTracing();
     SecretClient client = new SecretClientBuilder()
       .endpoint("<your-vault-url>")
       .credential(new DefaultAzureCredentialBuilder().build())
       .buildClient();
+   doClientWork(client);
+}
 
-    Span span = TRACER.spanBuilder("user-span").startSpan();
-    try (Scope scope = TRACER.withSpan(span)) {
-        // Thread bound (sync) calls will automatically pick up the parent span and you don't need to pass it explicitly.
-        secretClient.setSecret(new Secret("secret_name", "secret_value"));
-    } finally {
-        span.end();
-    }
+@WithSpan("my-span")
+public static void doClientWork(SecretClient client) {
+    // WithSpan annotation creates a parent span and makes it current, which propagates into synchronous calls
+    // automatically. ApplicationInsights agent or OpenTelemetry agent also propagate context through async reactor calls.
+    // When manually instrumenting without agent help, please follow the next example for async context propagation.
+    secretClient.setSecret(new Secret("secret_name", "secret_value"));
+}
+```
+
+When using async clients without Application Insights Java agent or OpenTelemetry agent, please do context propagation manually:
+
+```java
```

Comment:
Is it possible to inject these code snippets from a compilable source as we do for other client libraries' READMEs? It used to have circular dependency issues with the old snippet tool but the new tool may support this.